pkg/plugin/datasource.go

```diff
@@ -130,27 +129,34 @@ func (ds *ODataSource) query(clientInstance ODataClient, query backend.DataQuery
 		return response
 	}
 
-	timeProperty := qm.TimeProperty.Name
+	// Prevent empty queries from being executed
+	if qm.TimeProperty == nil && len(qm.Properties) == 0 {
+		return response
+	}
+
 	frame := data.NewFrame("response")
 	frame.Name = query.RefID
 	if frame.Meta == nil {
 		frame.Meta = &data.FrameMeta{}
 	}
 	frame.Meta.PreferredVisualization = data.VisTypeTable
-	labels, err := data.LabelsFromString("time=" + timeProperty)
```

Comment:
The setting of the labels should currently remain in place. It is intended in particular for the case that the time property is not called 'time'. This allows you to recognize at any time in Grafana which property is used for the time value, even if there is no property with the name 'time' or if several properties are suitable as time properties due to their data type.